app/src/main/java/com/humberto/tasky/agenda/presentation/agenda_details/AgendaDetailsViewModel.kt

```diff
@@ -1,32 +1,33 @@
 package com.humberto.tasky.agenda.presentation.agenda_details
 
+import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
-import com.humberto.tasky.main.navigation.AgendaDetails
+import com.humberto.tasky.agenda.presentation.AgendaItemType
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.update
 import javax.inject.Inject
 
 class AgendaDetailsViewModel @Inject constructor(
-
+    savedStateHandle: SavedStateHandle
 ): ViewModel() {
 
-    private val _agendaDetailsState = MutableStateFlow<AgendaDetailsState?>(null)
-    val agendaDetailsState: StateFlow<AgendaDetailsState?> = _agendaDetailsState.asStateFlow()
+    private val agendaItemId: String? = savedStateHandle.get<String>("agendaItemId")
+    private val agendaItemType: AgendaItemType = savedStateHandle.get<AgendaItemType>("agendaItemType")!!
```

Comment:
For your interest, if the type is missing here, the app will crash. Don’t you think a better UX would be to navigate back to the previous destination and tell the user something went wrong and that they need to try again?

This is not a requirement for Tasky but just an example of the more minor things you can add to a real-world project that might not have been explicitly stated in the requirements document but is expected by clients and will most likely be requested to be added later in the project’s development.

Reply:
I think the type can't be missing, that's why it's not nullable on AgendaDetails class since I'm always sending a type.

Reply:
Oh wow! I was trying to do something like that but wasn't able to. Awesome!